cleanlab/outlier.py

```diff
@@ -0,0 +1,534 @@
+
+        verbose : bool, default = True
+          Set to ``False`` to suppress all print statements.
+
+        """
+        _ = self._shared_fit(
+            features=features,
+            pred_probs=pred_probs,
+            labels=labels,
+            verbose=verbose,
+        )
+
+    def score(
+        self, *, features: Optional[np.ndarray] = None, pred_probs: Optional[np.ndarray] = None
+    ) -> np.ndarray:
+        """
+        Uses fitted estimator and passed in `features` or `pred_probs` to calculate out-of-distribution scores with
+        optional params passed in during object initialization. Score for each example that roughly corresponds to the
+        likelihood this example stems from the same distribution as the dataset features (i.e. is not an outlier).
+        Scores lie in [0,1] with smaller values indicating examples that are less typical under the dataset distribution
+        (values near 0 indicate outliers).
+
+        If `features` are passed, outlier score for each example based on its feature values is returned.
+        If `pred_probs` are passed in out of distribution score for each example based on its `pred_prob` values is returned.
+
+        Parameters
+        ----------
+        features : np.ndarray, optional
+          Feature array of shape ``(N, M)``, where N is the number of examples and M is the number of features used to represent each example.
+          For details, see `features` in :py:func:`fit <cleanlab.outlier.OutOfDistribution.fit>` function.
+
+        pred_probs : np.ndarray, optional
+          An array of shape ``(N, K)`` of model-predicted probabilities.
+          For details, see `pred_probs` in :py:func:`fit <cleanlab.outlier.OutOfDistribution.fit>` function.
+
+        Returns
+        -------
+        scores : np.ndarray
+          If `features` are passed in, `ood_features_scores` are returned.
+          The score is based on the average distance between the example and its K nearest neighbors in the dataset
+          (in feature space).
+
+          If `pred_probs` are passed in, `ood_predictions_scores` are returned.
+          The score is based on the model predicted probabilities.
+        """
+        self._assert_valid_inputs(features, pred_probs)
+
+        if features is not None:
+            if self.params["knn"] is None:
+                raise ValueError(
+                    f"OOD Object needs to be fit on features first. Call fit() or fit_scores() before this function."
+                )
+            else:
+                scores, _ = _get_ood_features_scores(
+                    features, **self._get_params(self.OUTLIER_PARAMS)
+                )
+
+        if pred_probs is not None:
+            if self.params["confident_thresholds"] is None and self.params["adjust_pred_probs"]:
+                raise ValueError(
+                    f"OOD Object needs to be fit on features first. Call fit() or fit_scores() before this function."
```

Comment:
better error message:

OOD estimator needs to be fit on `pred_probs` first since `params["adjust_pred_probs"]=True`